cmake/singularity/hdf5.cmake

```diff
@@ -0,0 +1,28 @@
+macro(singularity_enable_hdf5 target)
+  # CMake's FindHDF5 package ignores any path in CMAKE_PREFIX_PATH that
+  # doesn't contain the hdf5-config.cmake file in a certain location.
+  # Some of our HDF5 builds don't have this.  So we have to give CMake
+  # a little extra help...
+  if(NOT HDF5_ROOT)
+    find_path(HDF5_INCLUDE_DIR hdf5.h
+      HINTS ENV HDF5_ROOT
+      PATH_SUFFIXES include)
+    get_filename_component(HDF5_ROOT ${HDF5_INCLUDE_DIR} DIRECTORY)
+  endif()
+
+  find_package(HDF5 COMPONENTS HL REQUIRED)
+
+
+  target_include_directories(${target} SYSTEM PUBLIC ${HDF5_INCLUDE_DIRS})
+  target_link_libraries(${target} PUBLIC ${HDF5_LIBRARIES})
+
+  if(HDF5_IS_PARALLEL)
+    find_package(MPI COMPONENTS CXX REQUIRED)
+    target_link_libraries(${target} PUBLIC MPI::MPI_CXX)
+  endif()
```

Comment:
I can imagine this being a little finicky: If a user has both parallel and serial MPI installed and wants to specify one or the other... is that on us?

Reply:
The `MPI` requirement is solely for `HDF5`. If the `HDF5` that is located was built with `MPI`, then we need to bring in the library, otherwise there will be build errors.

Reply:
oh yeah I had a brain fart. I meant if the user has serial vs parallel HDF5 installed and wants to specify one or the other. I guess there's cmake flags for that right? That the user can specify?